src/components/tiptap/MenuBar.tsx

```diff
@@ -0,0 +1,165 @@
+
+import React, { Fragment } from 'react'
+import MenuItem from './MenuItem'
+
+type EditorProps = {
+  editor: any; 
+};
+type MenuItemProps =  {
+  icon: JSX.Element;
+  title: string;
+  action: () => void;
+  isActive: (() => boolean) | null | undefined;
+};
+
+type Item  = MenuItemProps & {
+  type?: string | undefined;
+};
+const MenuBar: React.FC<EditorProps> = ({ editor }) => {
+  const defaultIsActive = (): boolean => false;
+  const items: Item[]=[
+    {
+      icon: <path d="M8 11H12.5C13.8807 11 15 9.88071 15 8.5C15 7.11929 13.8807 6 12.5 6H8V11ZM18 15.5C18 17.9853 15.9853 20 13.5 20H6V4H12.5C14.9853 4 17 6.01472 17 8.5C17 9.70431 16.5269 10.7981 15.7564 11.6058C17.0979 12.3847 18 13.837 18 15.5ZM8 13V18H13.5C14.8807 18 16 16.8807 16 15.5C16 14.1193 14.8807 13 13.5 13H8Z" fill="rgba(255,255,255,1)"></path>,
```

Comment:
❌ 
SVG icons should be in separated files

Reply:
you should do the same for all the icons

Reply:
also, these icons are static assets, they should live in the public folder (read nextJs docs on static assets) 
and will pass only the relative path to them in here, 

read a bit about how to handle static assets in react/nextjs apps, specifically icons and SVGs

Reply:
i have deleted this and put each svg in a file into the public/icons and then imported the svg files to use as an image using Image/next